### PR TITLE
D kent

### PR DESCRIPTION
**Pull request by icegriffinguru**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
     "react-router-dom": "6.9.0",
     "react-scripts": "5.0.1",
     "web-vitals": "3.3.1",
-    "yup": "1.1.1"
+    "yup": "1.1.1",
+    "zustand": "^4.3.8"
```
Comment (damienen) [inline package.json:36]: Change this to "zustand": "4.3.8" (better to have a stable version)

File: src/AppHarness/StoreProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { useAccountStore, useMarketStore, useMintStore } from "store";
+import { useChainMeta } from "store/ChainMetaContext";
+import { useUser } from "store/UserContext";
+import { DataNftMintContract } from "MultiversX/dataNftMint";
+
+export const StoreProvider = ({ children }: PropsWithChildren) => {
+  const { chainID } = useGetNetworkConfig();
+  const { address } = useGetAccountInfo();
+  const { hasPendingTransactions } = useGetPendingTransactions();
+  const { chainMeta } = useChainMeta();
```
Comment (damienen) [inline src/AppHarness/StoreProvider.tsx:20]: Won't this be called whenever a new page is rendered? Shouldn't these be saved in the store itself, too at login?

**Review by damienen — COMMENTED** [review]
I like what I am seeing. I'd like @fuzzyduke to test if there are performance improvements, too

Left some minor comments, please check them out, @icegriffinguru . Else, stuff looks good!

File: src/AppHarness/StoreProvider.tsx (hunk 2 above)
Comment (icegriffinguru) [reply src/AppHarness/StoreProvider.tsx:20]: Okay

File: package.json (hunk 1 above)
Comment (icegriffinguru) [reply package.json:36]: Okay

**Pull request closed without merging**